packages/eas-cli/src/project/publish.ts

```diff
@@ -858,9 +873,119 @@ export function getRuntimeToPlatformsAndFingerprintInfoMappingFromRuntimeVersion
           runtimeVersionInfoObjects.map(
             runtimeVersionInfoObject => runtimeVersionInfoObject.runtimeVersionInfo.fingerprint
           )[0] ?? null,
+        fingerprintHash:
+          runtimeVersionInfoObjects.map(
+            runtimeVersionInfoObject => runtimeVersionInfoObject.runtimeVersionInfo.fingerprintHash
+          )[0] ?? null,
+      };
+    }
+  );
+}
+
+export async function maybeCalculateFingerprintForRuntimeVersionInfoObjectsWithoutExpoUpdatesAsync({
```

Comment:
nit: the logic of this function is kind of complicated. would be good to split as smaller functions and add some unit test.
but fine if that requires too much effort.